__tests__/dataframe.test.ts

```diff
@@ -2144,6 +2144,225 @@ describe("io", () => {
     const actual = df.toRecords();
     expect(JSON.stringify(actual)).toEqual(JSON.stringify(expected));
   });
+  test("toRecords:datetime:microseconds:fromIPC", () => {
+    // eslint-disable-next-line @typescript-eslint/no-require-imports
+    const path = require("node:path");
+    const tmpPath = path.resolve(__dirname, "./test_datetime_us.ipc");
+
+    try {
+      const dt1 = new Date("2024-02-28T14:53:00.000Z");
+      const dt2 = new Date("2025-01-03T09:30:00.000Z");
+      const dt3 = new Date("2024-12-31T23:59:59.999Z");
+
+      const df = pl.DataFrame({
+        id: [1, 2, 3],
+        datetime_us: pl.Series(
+          "datetime_us",
+          [dt1, dt2, dt3],
+          pl.Datetime("us"),
+        ),
+        date_field: pl.Series("date_field", [dt1, dt2, dt3], pl.Date),
+      });
+
+      df.writeIPC(tmpPath);
+      const dfRead = pl.readIPC(tmpPath);
+
+      const variants = dfRead.dtypes.map((dt) => dt.variant);
+      expect(variants[0]).toMatch(/Int64|Float64/);
+      expect(variants[1]).toBe("Datetime");
+      expect(variants[2]).toBe("Date");
+
+      const records = dfRead.toRecords() as unknown as Array<{
+        id: number;
+        datetime_us: Date;
+        date_field: Date;
+      }>;
+
+      expect(records).toHaveLength(3);
+
+      const date1 = records[0].datetime_us;
```

Comment:
Why test each element here

Reply:
I do not have a preference, but testing each element of Date seems bit much when a Date object is created below.